src/Native/Unix/System.Native/pal_io.h

```diff
@@ -32,6 +32,7 @@ typedef struct
     int64_t BirthTimeNsec; // nanosecond part
     int64_t Dev;       // ID of the device containing the file
     int64_t Ino;       // inode number of the file
+    uint32_t UserFlags; // user defined flags
```

Comment:
This is going to be problematic.

Because of how we've moved code around and shared it between the coreclr and corefx repos, some code in System.Private.CoreLib in the coreclr repo depends on System.Native.so/dylib and this type.  When we add this field here, we'll end up corrupting data in System.Private.CoreLib until it can be updated accordingly, but that can't happen until these changes are merged.  Catch-22.

I see two immediate ways around that:
1) Temporarily introduce a new set of APIs/structs that this code uses, e.g. Xx2 instead of Xx.  Then once everything has flown to coreclr and it's been updated accordingly and those bits consumed back into corefx, the old ones can be deleted. Then repeat the process to rename everything back to the "good" names.
2) Try to work within the confines of the existing struct as it's defined, e.g. use the Flags field (which we introduced as part of the PAL and so can do with it what we want) to pass the relevant data rather than defining a new UserFlags field.

How would things look if we did (2)?  Is it possible?  Could it be done in a relatively clean way, or would it be messy?

cc: @jkotas, @danmosemsft (longer term we may want to rethink where the shared shims live)

Reply:
> x2 instead of Xx.

We do have x2 APIs for this today because of similar type of change some time ago. We can use this to go back to APIs without the `2` suffix.

> I see two immediate ways around that:

3rd option would be to first add the field and let the change propagate everywhere. Once the field is everywhere start actually using it. 

This is not that different from how any changes on the CoreCLR/CoreFX boundaries need to be staged, e.g. changes of public APIs that are in use by CoreFX have to be often staged in similar way.

Reply:
> 3rd option

Ah, right.  I like that option, too.

> We do have x2 APIs for this today because of similar type of change some time ago. We can use this to go back to APIs without the 2 suffix.

You're right.  Let's do that, then.

Reply:
If I understood it correctly I should extract this change to separate PR and this PR will be merged once it is propagated everywhere, correct?

Reply:
Yes @davidkaya I believe htat's right. (We didn't notice you were waiting on this q)

Reply:
Yup, we first add the field, and then add the rest subsequently.